index.js

```diff
@@ -911,30 +913,36 @@ class Watcher extends EventEmitter {
     if (this.closed) return
     if (this.opened === false) await this._opening
 
-    const snapshot = this.bee.snapshot()
-    this.stream = snapshot.createDiffStream(this.latestDiff, this.range)
+    if (this.current) await this.current.close()
```

Comment:
Not sure if this matters in practice, but could run in async parallel:
```
const closeProms = []
if (this.current) closeProms.push(this.current.close())
if (this.previous) closeProms.push(this.previous.close())
await Promise.all(closeProms)
```
Or could perhaps also just launch and forget both, like
`if (this.current) this.current.close().catch(safetycatch)`

Reply:
Yep, we thought the same, I was thinking about `Promise.allSettled` so it guards against errors but it still processes both